canvas/canvas_course_usage.py

```diff
@@ -0,0 +1,109 @@
+import logging
+import time
+import pandas as pd
+from json.decoder import JSONDecodeError
+from requests_futures.sessions import FuturesSession
+from concurrent.futures import as_completed
+import json
+logger = logging.getLogger(__name__)
+
+
+class CanvasCourseUsage:
+    def __init__(self, canvas_url, canvas_token, retry_attempts, course_ids):
+        self.canvas_url = canvas_url
+        self.canvas_token = canvas_token
+        self.course_ids = course_ids
+        self.retry_attempts = retry_attempts
+        self.canvas_usage_courses = []
+        self.course_retry_list = []
+        self.retry_count = 0
+
+    def parsing_canvas_course_usage_data(self, response) -> None:
+        logger.info("parsing_canvas_course_usage_data Call")
```

Comment:
We could also potentially make this `debug`. There are just going to be a lot of these, and it seems like it would be most useful when changing code.